### PR TITLE
[Hexagon] Use LiveRegUnits

### PR DESCRIPTION
**Pull request by AZero13**

### CODE REVIEW

File: llvm/lib/Target/Hexagon/HexagonGenMux.cpp [hunk 1]
```diff
@@ -346,11 +346,11 @@ bool HexagonGenMux::genMuxInBlock(MachineBasicBlock &B) {
 
   // Fix up kill flags.
 
-  LivePhysRegs LPR(*HRI);
+  LiveRegUnits LPR(*HRI);
   LPR.addLiveOuts(B);
   auto IsLive = [&LPR, this](unsigned Reg) -> bool {
```
Comment (kparzysz) [inline llvm/lib/Target/Hexagon/HexagonGenMux.cpp:351]: With this change, just replace `IsLive` with `LPR.available`.

**Comment by kparzysz** [comment]
LGTM.  You can remove the `IsLive` definition, as it's not used anymore.  Thanks.

**Comment by AtariDreams** [comment]
> LGTM. You can remove the `IsLive` definition, as it's not used anymore. Thanks.

Good now? Also I do not have commit permissions.

**Comment by AtariDreams** [comment]
> LGTM. You can remove the `IsLive` definition, as it's not used anymore. Thanks.

I cannot do that because tests break and report bad machine code.

**Comment by kparzysz** [comment]
Please stop using force-pushes.  They can mess up previous review comments.

**Comment by kparzysz** [comment]
There was a mistake in my comment: `IsLive` is equivalent to `!LPR.available`.  Please restore the previous change and apply this correction.

**Comment by AtariDreams** [comment]
> There was a mistake in my comment: `IsLive` is equivalent to `!LPR.available`. Please restore the previous change and apply this correction.

Done!

**Review by kparzysz — APPROVED** [review]
Please remove `IsLive`, LGTM with that change.

**Comment by AtariDreams** [comment]
> Please remove `IsLive`, LGTM with that change.

I tried but then tests started failing, so I cannot do that.

**Comment by kparzysz** [comment]
> I tried but then tests started failing, so I cannot do that.

The `IsLive` function checks if the register is in use (i.e. live).  We then use it to set the kill flag on it: if the register is not live, then the use of it that we see (we're going backwards here) is the last one, so it's a kill.  The `available` function in LiveRegUnits does exactly what `IsLive` is doing here.  In this context "available" means "not in use", i.e. not live.

Is you replace the use of `IsLive` with `!available` it should work.

**Comment by AtariDreams** [comment]
> > I tried but then tests started failing, so I cannot do that.
> 
> The `IsLive` function checks if the register is in use (i.e. live). We then use it to set the kill flag on it: if the register is not live, then the use of it that we see (we're going backwards here) is the last one, so it's a kill. The `available` function in LiveRegUnits does exactly what `IsLive` is doing here. In this context "available" means "not in use", i.e. not live.
> 
> Is you replace the use of `IsLive` with `!available` it should work.

Done! @kparzysz

**Review by kparzysz — APPROVED** [review]
LGTM

**Pull request merged**